algviz/interface/visitors.py

```diff
@@ -190,3 +191,59 @@ class WidgetVisitor(Visitor):
 
     def visit(self, *args, **kwargs):
         return super().visit(*args, **kwargs)
+
+@default_for_type(type(None))
+class NullVisitor(Visitor):
+    type_ = Tokens.NULL_T
+
+    def uid(self, item):
+        return structures.Null.uid
+
+    def visit(self, *args, **kwargs):
+        return super().visit(*args, **kwargs)
+
+class TreeVisitor(Visitor):
```

Comment:
I would consider adding a lot of these methods that act on trees to the tree data structure itself, so that they can be accessed outside of the visitor.

Reply:
I think I've miscommunicated about what these visitors are for.  The user has to make a subclass that can visit their own (user-defined) data structure.  Our job is to avoid making assumptions about their data structure's API.  By calling a method, we'd be assuming that the method existed and did what we intended.